tests/experimental/test_mcmc.py

```diff
@@ -293,3 +302,56 @@ def test_history_keys(sampler: cuqi.experimental.mcmc.SamplerNew):
 
     # Check if the actual keys match the expected keys
     assert actual_keys == expected_keys, f"History keys mismatch. Expected: {expected_keys}, Actual: {actual_keys}"
+
+# Dictionary to store keys that are not expected to be updated after warmup.
+# Likely due to not implemented feature in the sampler.
+state_exception_keys = {
```

Comment:
(this is not directly relevant to this PR) it seems all these classes allow warmup but tuning is not implemented yet (thus scale is not updated). I suggest we add an item in issue #378 to say :

- [ ] For all the sampler classes in which tune is implemented as 'pass', do we need warning or error? to let the user know tuning is not happening?